tasks/internal/filter-already-released-advisory-images-task/filter-already-released-advisory-images-task.yaml

```
@@ -0,0 +1,152 @@
+        echo -n "$(context.taskRun.name)" > "$(results.internalRequestTaskRunName.path)"
+
+        exitfunc() {
+          local err=$1
+          local line=$2
+          local cmd=$3
+          if [ "$err" -eq 0 ]; then
+            echo -n "Success" > "$(results.result.path)"
+          else
+            echo -n "$0: ERROR '$cmd' failed at line $line - exited with status $err" > "$(results.result.path)"
+            if [ -f "$STDERR_FILE" ]; then tail -n 20 "$STDERR_FILE" >> "$(results.result.path)"; fi
+          fi
+          exit 0
+        }
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        ORIGIN="$(params.origin)"
+        ADVISORY_BASE_DIR="data/advisories/${ORIGIN}"
+
+        cd /tmp
+
+        echo "Cloning $GIT_REPO..."
+        git clone "$GIT_REPO" repo
+        cd repo
+
+        echo "Checking existing advisories in ${ADVISORY_BASE_DIR}..."
+        EXISTING_ADVISORIES=""
+        if [ -d "${ADVISORY_BASE_DIR}" ]; then
+          EXISTING_ADVISORIES=$(
+            find "${ADVISORY_BASE_DIR}" -mindepth 2 -type d -printf "%T@ %p\n" |
+            sort -nr | cut -d' ' -f2- | sed "s|^${ADVISORY_BASE_DIR}/||"
+          )
+        fi
+
+        CONTENT_IMAGES=$(jq -c '.components' <<< "$SNAPSHOT_JSON")
+
+        if [[ -z "$EXISTING_ADVISORIES" ]]; then
+          echo "No existing advisories found. No components have been released yet."
+          UNRELEASED_COMPONENTS=$(jq -r '.[].name' <<< "$CONTENT_IMAGES" | jq -R -s -c 'split("\n")[:-1]')
+          echo -n "$UNRELEASED_COMPONENTS" > "$(results.unreleased_components.path)"
+          echo -n "Success" > "$(results.result.path)"
+          exit 0
+        fi
+
+        for ADVISORY_SUBDIR in $EXISTING_ADVISORIES; do
+          ADVISORY_FILE="${ADVISORY_BASE_DIR}/${ADVISORY_SUBDIR}/advisory.yaml"
+          EXISTING_CONTENT=$(yq -o=json '.spec.content.images // []' "$ADVISORY_FILE")
+
+          echo "Comparing against: $ADVISORY_FILE"
+
+          CONTENT_IMAGES=$(echo "$CONTENT_IMAGES" | jq --argjson existing "$EXISTING_CONTENT" '
+            map(select(
+              .containerImage as $ci |
+              .tags as $tags |
+              .repository as $repo |
+              ($existing | map(select(
+                .containerImage == $ci and .tags == $tags and .repository == $repo
+              )) | length == 0)
+            ))')
+
+          # If after filtering, no images are left, then we can exit early
+          if jq -e 'length == 0' <<< "$CONTENT_IMAGES" >/dev/null; then
+            echo "All images in the snapshot have already been released in advisories. Stopping pipeline."
+            echo -n "[]" > "$(results.unreleased_components.path)"
+            echo -n "Success" > "$(results.result.path)"
+            exit 0
+          fi
+        done
+
+        ORIGINAL_COUNT=$(jq '.components | length' <<< "$SNAPSHOT_JSON")
+        NEW_COUNT=$(echo "$CONTENT_IMAGES" | jq 'length')
```

Why not 
```suggestion
        NEW_COUNT=$(jq 'length' <<< "$CONTENT_IMAGES")
```
I'm surprised checkton doesn't complain about this :)

---

Thanks, fixed as suggested!